apps/server/src/modules/video-conference/service/video-conference.service.spec.ts

```diff
@@ -332,64 +328,160 @@ describe('VideoConferenceService', () => {
+				const conferenceScope = VideoConferenceScope.COURSE;
+
+				authorizationService.hasPermission.mockReturnValueOnce(false).mockReturnValueOnce(true);
+				authorizationService.getUserWithPermissions.mockResolvedValueOnce(user);
+				courseService.findById.mockResolvedValueOnce(entity);
+
+				return {
+					user,
+					userId: user.id,
+					entity,
+					entityId: entity.id,
+					conferenceScope,
+				};
+			};
+
+			it('should call the correct authorisation order', async () => {
+				const { user, entity, userId, conferenceScope, entityId } = setup();
+
+				await service.determineBbbRole(userId, entityId, conferenceScope);
+
+				expect(authorizationService.hasPermission).toHaveBeenNthCalledWith(
+					1,
+					user,
+					entity,
+					AuthorizationContextBuilder.read([Permission.START_MEETING])
+				);
+				expect(authorizationService.hasPermission).toHaveBeenNthCalledWith(
+					2,
+					user,
+					entity,
+					AuthorizationContextBuilder.read([Permission.JOIN_MEETING])
+				);
+			});
+
 			it('should return BBBRole.VIEWER', async () => {
 				const { userId, conferenceScope, entityId } = setup();
-				authorizationService.hasPermissionByReferences.mockResolvedValueOnce(false);
-				authorizationService.hasPermissionByReferences.mockResolvedValueOnce(true);
 
-				const result: BBBRole = await service.determineBbbRole(userId, entityId, conferenceScope);
+				const result = await service.determineBbbRole(userId, entityId, conferenceScope);
 
 				expect(result).toBe(BBBRole.VIEWER);
-				expect(authorizationService.hasPermissionByReferences).toHaveBeenCalledWith(
-					userId,
-					AuthorizableReferenceType.Course,
-					entityId,
-					AuthorizationContextBuilder.read([Permission.JOIN_MEETING])
-				);
 			});
 		});
 
 		describe('when user has neither START_MEETING nor JOIN_MEETING permission', () => {
+			const setup = () => {
+				const user = userFactory.buildWithId();
+				const entity = courseFactory.buildWithId();
+				const conferenceScope = VideoConferenceScope.COURSE;
+
+				authorizationService.hasPermission.mockReturnValueOnce(false).mockReturnValueOnce(false);
+				authorizationService.getUserWithPermissions.mockResolvedValueOnce(user);
+				courseService.findById.mockResolvedValueOnce(entity);
+
+				return {
+					user,
+					userId: user.id,
+					entity,
+					entityId: entity.id,
+					conferenceScope,
+				};
+			};
+
 			it('should throw a ForbiddenException', async () => {
 				const { userId, conferenceScope, entityId } = setup();
-				authorizationService.hasPermissionByReferences.mockResolvedValueOnce(false);
-				authorizationService.hasPermissionByReferences.mockResolvedValueOnce(false);
 
-				const func = () => service.determineBbbRole(userId, entityId, conferenceScope);
-
-				await expect(func).rejects.toThrow(new ForbiddenException(ErrorStatus.INSUFFICIENT_PERMISSION));
+				await expect(service.determineBbbRole(userId, entityId, conferenceScope)).rejects.toThrow(
```

Comment:
I like the previous seperation of AAA because of more readability

Reply:
ok i change it back